sycl/test/abi/symbol_size.cpp

```diff
@@ -0,0 +1,44 @@
+// RUN: %clangxx -fsycl %s -o %t && %t
+
+#include <CL/sycl.hpp>
+#include <cassert>
+
+using namespace cl::sycl;
+
+#define CHECK_LAYOUT(class_name, size)                                    \
+  if (sizeof(class_name) != size) {                                       \
+    std::cout << "Size of class " << #class_name << " has changed. Was: " \
+              << #size << ". Now: " << sizeof(class_name) << std::endl;   \
+    HasChanged = true;                                                    \
+  }
+
+int main() {
+
+  bool HasChanged = false;
+
+  using accessor_t = accessor<int, 1, access::mode::read,
+                              access::target::global_buffer, access::placeholder::true_t>;
+  CHECK_LAYOUT(accessor_t, 32)
+  CHECK_LAYOUT(buffer<int>, 40)
```

Comment:
How about "compile-only" test? E.g. https://github.com/llvm/llvm-test-suite/blob/master/ABI-Testsuite/test/basic/basic.x.

Reply:
Compile-time checks are much better, thanks for the suggestion. However, I decided to go with `static_assert` instead of macros. Does it appeal to you?

Reply:
LGTM. Thanks.